Modix.Services/UserInfo/UserInfoCommand.cs

```diff
@@ -0,0 +1,297 @@
+﻿#nullable enable
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Discord;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Modix.Data;
+using Modix.Data.Extensions;
+using Modix.Data.Models.Core;
+using Modix.Data.Models.Moderation;
+using Modix.Services.AutoRemoveMessage;
+using Modix.Services.Extensions;
+using Modix.Services.Images;
+using Modix.Services.Utilities;
+
+namespace Modix.Services.UserInfo
+{
+    public class UserInfoCommand : IRequest
+    {
+        public UserInfoCommand(IMessage message, ulong guildId, ulong userId)
+        {
+            Message = message;
+            GuildId = guildId;
+            UserId = userId;
+        }
+        public ulong UserId { get; }
+        public ulong GuildId { get; }
+        public IMessage Message { get; }
+    }
+
+    public class UserInfoCommandHandler : AsyncRequestHandler<UserInfoCommand>
+    {
+        private readonly ModixContext _modixContext;
+        private readonly IDiscordClient _discordClient;
+        private readonly IAutoRemoveMessageService _autoRemoveMessageService;
+        private readonly IImageService _imageService;
+        private readonly ILogger<UserInfoCommandHandler> _logger;
+        private readonly ModixConfig _modixConfig;
+        private readonly Stopwatch _stopwatch = new Stopwatch();
+
+        public UserInfoCommandHandler(ModixContext modixContext, IDiscordClient discordClient,
+            IAutoRemoveMessageService autoRemoveMessageService, IImageService imageService, ILogger<UserInfoCommandHandler> logger,
+            IOptions<ModixConfig> modixOptions)
+        {
+            _modixContext = modixContext;
+            _discordClient = discordClient;
+            _autoRemoveMessageService = autoRemoveMessageService;
+            _imageService = imageService;
+            _logger = logger;
+            _modixConfig = modixOptions.Value;
+        }
+
+        protected override async Task Handle(UserInfoCommand request, CancellationToken cancellationToken)
+        {
+            _stopwatch.Start();
+
+            _logger.LogDebug("Getting info for user {UserId} in guild {GuildId}", request.UserId, request.GuildId);
+
+            var guild = await _discordClient.GetGuildAsync(request.GuildId);
+
+            var discordUser = await guild.GetUserAsync(request.UserId);
+
+            _logger.LogDebug("Got user from Discord API");
+
+            var modixUser = await GetUserAsync(request.UserId, request.GuildId, cancellationToken);
+
+            if (discordUser is null
+                || modixUser is null)
```

Comment:
We should check if `discordUser` is `null` can we skip the `modixUser` lookup all together?